pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -405,8 +405,7 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 		}
 	}
 
-	if _, nodeDrainExcluded := machine.Annotations[capi.ExcludeNodeDrainingAnnotation]; !removed ||
-		(!nodeDrainExcluded && !drainingSucceededCondition.IsTrue(machine) && drainingSucceededCondition.GetReason(machine) != capi.DrainingFailedReason) {
+	if !removed {
```

Comment:
Can you elaborate a little bit on why we can drop the check for annotation and status?

Reply:
We drop the check here because we actually want to reenqueue now regardless of whether or not we are draining, because CAPI will automagically force delete the node for us if all if the `NodeDrainTimeout` is reached.